pkg/config/validation/validation.go

```diff
@@ -798,6 +798,12 @@ var ValidateEnvoyFilter = registerValidateFunc("ValidateEnvoyFilter",
 			return nil, err
 		}
 
+		// If workloadSelector is defined and labels are not set, it is most likely
+		// an user error. Marking it as a warning to keep it backwards compatible.
+		if rule.WorkloadSelector != nil && len(rule.WorkloadSelector.Labels) == 0 {
```

Comment:
I think this can be checked in `validateAlphaWorkloadSelector` to apply to all resources

Reply:
good point

Reply:
@hzxuzhonghu I did consider that initially. But the above function does not pass back warning, but would need additional re-factoring.

Reply:
@hzxuzhonghu @ramaraochavali was able to refactor it. Ptal when you get a chance.